### PR TITLE
Adjusts UI for larger board and for colors

### PR DESCRIPTION
**Pull request by saralein**
Adjusts the display board to handle the larger board options.  Adds colors for prompts, the board, tokens, errors.

### CODE REVIEW

**Comment by zdavy** [comment]
Can we get some pictures some pictures attached to this PR so we can see what the new colors look like?

**Comment by saralein** [comment]
screen shot 2017-09-07 at 1 59 25 pm
screen shot 2017-09-07 at 2 01 17 pm
screen shot 2017-09-07 at 2 02 24 pm
screen shot 2017-09-07 at 2 02 45 pm
screen shot 2017-09-07 at 2 24 51 pm

**Comment by zdavy** [comment]
Very nice color scheme 🍬

File: lib/ui/string.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def blue
+    colorize(34)
+  end
+
+  def yellow
+    colorize(33)
+  end
+end
```
Comment (zdavy) [inline lib/ui/string.rb]: ruby makes it super easy to extend any existing object, and this is a very ruby way to solve this problem, but I'd like you to solve this using some sort of presentation layer. It's a little jarring to see `'foo'.blue`, it doesn't fit my worldview of a string literal.

File: spec/user_interface_spec.rb [hunk 2]
```diff
@@ -29,25 +37,37 @@
     end
   end
 
+  describe 'assign_color' do
+    it 'sets X to magenta and O to green' do
+      expect(user_interface.instance_variable_get(:@colors)).to eql({"X"=>:magenta, "O"=>:green})
```
Comment (damonkelley) [inline spec/user_interface_spec.rb]: Is there a way to test with without using `instance_variable_get`?

File: spec/user_interface_spec.rb [hunk 3]
```diff
@@ -6,21 +6,29 @@
...
-  let(:empty_string) { '  |   |  ' }
-  let(:full_string) { 'X | X | O' }
+  let(:moves_taken) { ['O', 'X', '-', '-', 'O', '-', 'X', 'X', 'O'] }
+  let(:header) {"\e[34m     Board     \e[0m           \e[34mMoves\e[0m"}
+  let(:empty_string) { "   \e[33m | \e[0m   \e[33m | \e[0m   " }
+  let(:full_string) { "\e[35m X \e[0m\e[33m | \e[0m\e[35m X \e[0m\e[33m | \e[0m\e[32m O \e[0m" }
+  let(:normal_move_string) { " 1 \e[33m | \e[0m 2 \e[33m | \e[0m 3 " }
+  let(:dim_move_string) { "\e[2m 1 \e[22m\e[33m | \e[0m\e[2m 2 \e[22m\e[33m | \e[0m 3 " }
+  let(:row_divider) { "\e[33m--- + --- + ---\e[0m" }
+  let(:output) { "\e[34m     Board     \e[0m           \e[34mMoves\e[0m\n\n#{empty_string}          1 \e[33m | \e[0m 2 \e[33m | \e[0m 3 \n#{row_divider}      #{row_divider}"}
```
Comment (damonkelley) [inline spec/user_interface_spec.rb]: Adding these ANSI codes into the expected messages might make the test more brittle. Maybe there is a way to keep these things separate?

File: lib/ui/string.rb (hunk 1 above)
Comment (saralein) [reply lib/ui/string.rb]: Made it into a [colorizer class](https://github.com/saralein/tic-tac-toe_ruby/pull/18/files#diff-025f4e3018e7f6d7253cde553a91e93bR1).

File: spec/user_interface_spec.rb (hunk 3 above)
Comment (saralein) [reply spec/user_interface_spec.rb]: I made a [mock colorizer](https://github.com/saralein/tic-tac-toe_ruby/pull/18/files#diff-7aa5a624018bd299e043383e4af27151R4) which just returns the string.

File: spec/user_interface_spec.rb (hunk 2 above)
Comment (saralein) [reply spec/user_interface_spec.rb]: I wondered if I should even test that.  I added a bit to some [tests](https://github.com/saralein/tic-tac-toe_ruby/pull/18/files#diff-7aa5a624018bd299e043383e4af27151R162) to indirectly see if it's working by checking that colors were called when X or O wins.

File: lib/players/ai.rb [hunk 4]
```diff
@@ -9,7 +9,8 @@ def initialize(board, checker, token, min_token)
   end
 
   def get_move(message, turns_remaining)
-    minimax(turns_remaining, -Float::INFINITY, Float::INFINITY, true)[1] + 1
+    depth = [turns_remaining, 9].min
```
Comment (zdavy) [inline lib/players/ai.rb]: What's the reason for this depth min

File: lib/ui/colorizer.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class Colorizer
```
Comment (zdavy) [inline lib/ui/colorizer.rb:1]: 💯

File: lib/players/ai.rb (hunk 4 above)
Comment (saralein) [reply lib/players/ai.rb]: I was trying to speed up the minimax on the larger 4x4 board by having it use a shallower depth until more turns have been taken.

Comment (zdavy) [reply lib/players/ai.rb]: If it's for the 4x4, It should be included in a 4x4 board PR. If something goes wrong and you have to revert, we'll have this orphan implementation detail hanging around.

Comment (saralein) [reply lib/players/ai.rb]: Makes sense.  Removed it from this branch.

File: spec/mocks/mock_colorizer.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+class MockColorizer
```
Comment (zdavy) [inline spec/mocks/mock_colorizer.rb:1]: also 💯

**Comment by zdavy** [comment]
:+1:

**Pull request merged**